build_deploy.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -e
+
+IMAGE_BUILD="ekidd/rust-musl-builder:1.30.1"
+
+IMAGE="quay.io/app-sre/cincinnati"
+IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+DOCKERFILE_DEPLOY="dist/Dockerfile"
+
+# clean up
+rm -rf target
+
+docker run --rm -v $(pwd):/home/rust/src $IMAGE_BUILD cargo build --release
+
+docker build -f $DOCKERFILE_DEPLOY -t "${IMAGE}:${IMAGE_TAG}" .
+
+if [[ -n "$QUAY_USER" && -n "$QUAY_TOKEN" ]]; then
+    DOCKER_CONF="$PWD/.docker"
+    mkdir -p "$DOCKER_CONF"
+    docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
```

Comment:
s/TOKEN/PASSWORD

Reply:
no, we do not have a password, since this is a robot acc we have only token. all our jobs are using this method to auth against quay and it perfectly works.

Reply:
It works fine. It's just a bit confusing to read since the terms are mixed up and it's technically a password:

```console
$ docker login --help | grep -- ' -p'
  -p, --password string   Password
```

Reply:
the thing is that actual creds are stored in vault and there is a TOKEN + username, and it is injected by CI for all jobs in the same way, we can't change ENV var name unfortunately here.

Reply:
I see, that's a pity but OTOH it's not critical. Thanks for explaining!